### PR TITLE
LPS-154043 DM File entry history - icons in dropdown

### PR DESCRIPTION
**Pull request by ambrinchaudhary**
https://issues.liferay.com/browse/LPS-154043

**Before** the change: 

Screenshot 2022-05-23 at 17 29 00

**After** the change: 
Screenshot 2022-05-23 at 17 19 32

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

**Comment by Hanlf** [comment]
Hi @ambrinchaudhary There is a semantic version failure should be related to this PR, could you take a look, thanks!

`     [exec] > Task :apps:document-library:document-library-api:baseline
     [exec]   PACKAGE_NAME                                       DELTA      CUR_VER    BASE_VER   REC_VER    WARNINGS  
     [exec] = ================================================== ========== ========== ========== ========== ==========
     [exec] * com.liferay.document.library.display.context       MAJOR      3.0.0      3.0.0      4.0.0      VERSION INCREASE REQUIRED
     [exec] 	<   class      com.liferay.document.library.display.context.BaseDLViewFileEntryHistoryDisplayContext
     [exec] 		+   method     getActionDropdownItems()
     [exec] 			+   return     java.util.List
     [exec] 	>   interface  com.liferay.document.library.display.context.DLViewFileEntryHistoryDisplayContext
     [exec] 		+   method     getActionDropdownItems()
     [exec] 			+   access     abstract
     [exec] 			+   return     java.util.List
     [exec] [Baseline Warning] Bundle Version Change Recommended: 12.0.0
     [exec] [Baseline Report] Mode: diff (persisted)
     [exec] Semantic versioning is incorrect while checking /opt/dev/projects/github/liferay-portal/tools/sdk/dist/com.liferay.document.library.api-11.0.1.jar against /opt/dev/projects/github/liferay-portal/.gradle/caches/modules-2/files-2.1/com.liferay/com.liferay.document.library.api/11.0.0/9d3b2e02a15d565185bc6b548cdc73a637fd7dcb/com.liferay.document.library.api-11.0.0.jar`

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/logic/UIItemsBuilder.java [hunk 1]
```diff
@@ -1038,6 +1083,29 @@ else if (_isMoveToTheRecycleBinActionAvailable()) {
...
+		return DropdownItemBuilder.putData(
+			"action", "deleteVersion"
+		).putData(
+			"deleteURL",
+			PortletURLBuilder.create(
+				_getActionURL(
+					"/document_library/edit_file_entry", Constants.DELETE,
+					_getRenderURL(
+						"/document_library/view_file_entry", _getRedirect()
+					).toString())
```
Comment (adolfopa) [inline modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/logic/UIItemsBuilder.java:1096]: This kind of chaining is not allowed by our SF rules. You'll either have to put the result of `_getRenderURL` in a variable and call toString on that, or use String.valueOf(_getRenderURL(...)).

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/logic/UIItemsBuilder.java [hunk 2]
```diff
@@ -1400,6 +1525,20 @@ public boolean isPublishActionAvailable() throws PortalException {
...
+			!_fileEntryDisplayContextHelper.hasUpdatePermission()) {
+
+			return false;
+		}
+
+		FileVersion latestFileVersion = _fileEntry.getLatestFileVersion();
+
+		String latestFileVersionVersion = latestFileVersion.getVersion();
+
+		return !latestFileVersionVersion.equals(_fileVersion.getVersion());
```
Comment (adolfopa) [inline modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/logic/UIItemsBuilder.java:1539]: In general, boolean expressions need to be used as part of a conditional statement a return an explicit true/false value. You'll need to change this to:
```
if (Objects.equals(latestFileVersion.getVersion(), _fileVersion.getVersion()) {
  return false;
}
else {
  return true;
}
```

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/logic/UIItemsBuilder.java (hunk 2 above)
Comment (ambrinchaudhary) [reply modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/logic/UIItemsBuilder.java:1539]: Ok, I will change that

**Comment by ambrinchaudhary** [comment]
ci:test:sf

**Comment by ambrinchaudhary** [comment]
ci:test:relevant

**Comment by ambrinchaudhary** [comment]
ci:test:sf

**Comment by ambrinchaudhary** [comment]
ci:test:relevant

**Review by AliciaGarciaGarcia — APPROVED** [review]

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Pull request closed without merging**